File: sample-push-notifications/sample-push-notifications/Authorization/RootParentVC.m
//
//  RootParentVC.m
//  sample-push-notifications
//
//  Created by Injoit on 1/29/20.
//  Copyright © 2020 Quickblox. All rights reserved.
//

#import "RootParentVC.h"
#import "SplashScreenVC.h"
#import "PushViewController.h"
#import "UIColor+Chat.h"

@interface PushesNavigationController: UINavigationController
@end
@implementation PushesNavigationController
@end

@interface RootParentVC ()
//MARK: - Properties
@property (nonatomic, strong) UIViewController *current;
@end

@implementation RootParentVC
//MARK: - Life Cycle
- (instancetype)init {
    self = [super init];
    if (self) {
        UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Authorization" bundle:nil];
        SplashScreenVC *splashScreenVC = [storyboard instantiateViewControllerWithIdentifier:@"SplashScreenVC"];
        self.current = splashScreenVC;
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];

    [self setupChildViewController:self.current];
}

//MARK - Setup
- (void)setupChildViewController:(UIViewController *)new {
    
    if ([self.current isEqual:new]) {
        [self addChildViewController:new];
        new.view.frame = self.view.bounds;
        [self.view addSubview:new.view];
        [new didMoveToParentViewController:self];

    } else {
        [self addChildViewController:new];
        new.view.frame = self.view.bounds;
        [self.view addSubview:new.view];
        [new didMoveToParentViewController:self];
        
        [self.current willMoveToParentViewController:nil];
        [self.current.view removeFromSuperview];
        [self.current removeFromParentViewController];
        self.current = new;
    }
}

- (UIStatusBarStyle) preferredStatusBarStyle {
    return UIStatusBarStyleLightContent;
}

// MARK: - Public Methods
- (void)showLoginScreen {
    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Authorization" bundle:nil];
    UINavigationController *new = [storyboard instantiateViewControllerWithIdentifier:@"AuthNavVC"];
    
    [self setupChildViewController:new];
}

- (void)showPushesScreen {
    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
    PushViewController *pushViewController = [storyboard instantiateViewControllerWithIdentifier:@"PushViewController"];
    
    PushesNavigationController *new = [[PushesNavigationController alloc] initWithRootViewController:pushViewController];
    new.navigationBar.barTintColor = [UIColor mainColor];
    new.navigationBar.barStyle = UIBarStyleBlack;
    new.navigationBar.shadowImage = UIImage.new;
    [new.navigationBar setTranslucent:NO];
    new.navigationBar.tintColor = UIColor.whiteColor;
    new.navigationBar.titleTextAttributes = @{ NSForegroundColorAttributeName:UIColor.whiteColor};
    
    [self setupChildViewController:new];
    
}

@end

File: sample-push-notifications-swift/sample-push-notifications-swift/Components/Authorization/RootParentVC.swift
//
//  RootParentVC.swift
//  sample-push-notifications-swift
//
//  Created by Injoit on 1/27/20.
//  Copyright © 2020 QuickBlox. All rights reserved.
//

import UIKit

class PushesNavigationController: UINavigationController { }

class RootParentVC: UIViewController {
    //MARK: - Properties
    var current: UIViewController
    
    //MARK: - Life Cycle
    init() {
        let storyboard = UIStoryboard(name: "Authorization", bundle: nil)
        current = storyboard.instantiateViewController(withIdentifier: "SplashScreenVC") as! SplashScreenVC
        super.init(nibName:  nil, bundle: nil)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        changeCurrentViewController(current)
    }
    
    // MARK: - Public Methods
    func showLoginScreen() {
        let storyboard = UIStoryboard(name: "Authorization", bundle: nil)
        let authNavVC = storyboard.instantiateViewController(withIdentifier: "AuthNavVC") as! UINavigationController
        
        changeCurrentViewController(authNavVC)
    }
    
    func showPushScreen() {
        let storyboard = UIStoryboard(name: "Main", bundle: nil)
        if let pushVC = storyboard.instantiateViewController(withIdentifier: "PushViewController") as? PushViewController {
            let pushScreen = PushesNavigationController(rootViewController: pushVC)
            pushScreen.navigationBar.barTintColor = #colorLiteral(red: 0.2216441333, green: 0.4713830948, blue: 0.9869660735, alpha: 1)
            pushScreen.navigationBar.barStyle = .black
            pushScreen.navigationBar.shadowImage = UIImage()
            pushScreen.navigationBar.isTranslucent = false
            pushScreen.navigationBar.tintColor = .white
            pushScreen.navigationBar.titleTextAttributes = [NSAttributedString.Key.foregroundColor: UIColor.white]
            
            changeCurrentViewController(pushScreen)
        }
    }
    
    //MARK: - Internal Methods
    private func changeCurrentViewController(_ newCurrentViewController: UIViewController) {
            addChild(newCurrentViewController)
            newCurrentViewController.view.frame = view.bounds
            view.addSubview(newCurrentViewController.view)
            newCurrentViewController.didMove(toParent: self)
            
            if current == newCurrentViewController {
                return
            }
            current.willMove(toParent: nil)
            current.view.removeFromSuperview()
            current.removeFromParent()
            current = newCurrentViewController
    }

    //MARK - Setup
    override var preferredStatusBarStyle: UIStatusBarStyle {
        return .lightContent
    }
}
